### PR TITLE
[Incubator][VC] ignore annotations that have specific prefixes 

### PR DESCRIPTION
**Pull request by charleszheng44**
Try to resolve issue #330 

when comparing virtual object and the real object, syncer will ignore annotation
prefixes specified in env `STDLN_OBJ_ANNO_PREFXS`

### CODE REVIEW

File: incubator/virtualcluster/config/setup/all_in_one.yaml [hunk 1]
```diff
@@ -238,6 +238,21 @@ spec:
           image: virtualcluster/syncer-amd64
           imagePullPolicy: Always
           name: vc-syncer
+          env:
+          - name: STDLN_OBJ_ANNO_PREFXS
```
Comment (zhuangqh) [inline incubator/virtualcluster/config/setup/all_in_one.yaml]: how about `STDIN_OBJ_ANNO_PREFIXES` ?

PREFXS  this abbreviation is confusing

File: incubator/virtualcluster/pkg/syncer/conversion/equality.go [hunk 2]
```diff
@@ -25,8 +26,25 @@ import (
...
+	// load the StandaloneObjAnnotationPrefixex from envs. When the syncer
+	// run in a pod, the env will be set through a configmap
+	// TODO should we add a feature to restart the syncer if the configmap
+	// is updated
+	soap := os.Getenv("STDLN_OBJ_ANNO_PREFXS")
+	if soap == "" {
+		return
+	}
+	// the delimiter of the prefix list is space
+	StandaloneObjAnnotationPrefixes = append(StandaloneObjAnnotationPrefixes, strings.Split(soap, " ")...)
```
Comment (zhuangqh) [inline incubator/virtualcluster/pkg/syncer/conversion/equality.go]: dedup and remove space-only prefix

may there is a continuous space.

File: incubator/virtualcluster/pkg/syncer/utils/string.go [hunk 3]
```diff
@@ -45,3 +48,13 @@ func GenAnAvailableName(names []string, prefix string, n int) string {
 		}
 	}
 }
+
+// HasPrefixs checks whether the string 's' begins with any prefix in 'prefixs'.
+func HasPrefixs(s string, prefixs []string) bool {
```
Comment (zhuangqh) [inline incubator/virtualcluster/pkg/syncer/utils/string.go]: HasPrefixes

File: incubator/virtualcluster/pkg/syncer/conversion/equality.go (hunk 2 above)
Comment (charleszheng44) [reply incubator/virtualcluster/pkg/syncer/conversion/equality.go]: use `strings.Field` instead

File: incubator/virtualcluster/pkg/syncer/utils/string.go (hunk 3 above)
Comment (charleszheng44) [reply incubator/virtualcluster/pkg/syncer/utils/string.go]: typo fixed

File: incubator/virtualcluster/config/setup/all_in_one.yaml (hunk 1 above)
Comment (charleszheng44) [reply incubator/virtualcluster/config/setup/all_in_one.yaml]: change to `STDLN_OBJ_ANNO_PREFIXES`

**Pull request merged**